assets/src/design-system/components/switch/index.js

```diff
@@ -0,0 +1,233 @@
+  border-radius: 100px;
+  color: ${({ theme }) => theme.colors.fg.primary};
+`;
+
+const HiddenRadioButton = styled.input.attrs({ type: 'radio' })`
+  position: absolute;
+  height: 0 !important;
+  width: 0 !important;
+  padding: 0 !important;
+  min-width: unset !important;
+  appearance: none;
+  border: 0 !important;
+  box-shadow: none !important;
+  margin: -1px !important;
+  outline: none !important;
+  overflow: hidden;
+`;
+
+const RadioButtonLabel = styled(Text).attrs({ forwardedAs: 'label' })`
+  box-sizing: border-box;
+  flex: 1;
+  width: 50%;
+  padding: 0px 6px;
+  z-index: 1;
+  color: ${({ isActive, theme }) =>
+    isActive ? theme.colors.inverted.fg.primary : theme.colors.fg.secondary};
+  cursor: pointer;
+  overflow: hidden;
+  text-align: center;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  transition: color 0.15s ease-out;
+
+  ${({ disabled }) =>
+    disabled &&
+    css`
+      cursor: default;
+      opacity: 0.3;
+    `}
+
+  /* add focus styling on the slider when the hidden input is focused */
+  :focus-within ~ span {
+    ${themeHelpers.focusCSS};
+  }
+`;
+
+const SlidingButton = styled.span`
+  box-sizing: border-box;
+  position: absolute;
+  display: block;
+  top: 0;
+  left: 0;
+  width: 50%;
+  height: ${SWITCH_HEIGHT}px;
+  border-radius: 100px;
+  background-color: ${({ theme }) => theme.colors.interactiveBg.primaryNormal};
+  transition: left 0.15s ease-out;
+  z-index: 0;
+
+  ${({ hasOffset }) => hasOffset && `left: 50%`}
+`;
+
+/**
+ * A controlled radio group that looks like a switch.
+ *
+ * Keyboard navigation:
+ * - [Tab] can be used to navigate between radio groups
+ * - [Enter], [Space], [ArrowLeft], [ArrowRight] [ArrowUp], and [ArrowDown] can be used to select radio buttons
+ *
+ * @param {Object} props The props
+ * @param {string} props.className className that is applied to the outer container
+ * @param {string} props.disabled whether the radio buttons are disabled
+ * @param {string} props.groupLabel the label for the radio group. Used for accessibility
+ * @param {string} props.id unique id used for accessibility
+ * @param {string} props.name name of the radio group
+ * @param {string} props.offLabel label for the 'off' radio button
+ * @param {Function} props.onChange change event handler
+ * @param {string} props.onLabel label for the 'on' radio button
+ * @param {boolean} props.value the value of the radio group
+ * @return {Object} The radio button
+ */
+export const Switch = forwardRef(function (
+  {
+    className = '',
+    disabled,
+    groupLabel,
+    id,
+    offLabel,
+    onChange,
+    onLabel,
+    value,
+    ...props
+  },
+  ref
+) {
+  const radioGroupRef = useRef(ref);
+  const ids = useMemo(
+    () => ({
+      group: id || uuidv4(),
+      offInput: uuidv4(),
+      onInput: uuidv4(),
+    }),
+    [id]
+  );
+
+  const handleChange = useCallback(
+    (evt) => {
+      onChange(evt, evt.target.value === 'true');
+    },
+    [onChange]
+  );
```

Comment:
I'm making the values of the radio buttons the booleans `true` and `false`. However, when they are rendered by react as html elements their values become the strings `"true"` and `"false"`.

The value on the event is a string, so we need a way to give the boolean value to the `onChange` event handler.

I don't _love_ this api, but I couldn't see any other way to do this. Thoughts?

Reply:
Yea I think I'm not used to making the values of a radio either `true` or `false`. I feel like there are aspects of this that make it feel between a checkbox and a radio because in a sense we're trying to map checked/unchecked to two radio buttons.

I can't think of a better way to do it as a radio tho 🤷

Reply:
Would there be any benefit to making it a checkbox? I feel like the arrow keys toggling it might make it weird at that point tho. hmph 🤔

Reply:
To me a switch is a non-boolean checkbox (meaning the 2 values could be 1 and 2 or tacos and enchiladas rather than just a true/false state - but only because of values tied to accessibility), which makes it into a 2 input radio group. I think "traditional" radio input values aren't true/false and also expect more than 2 options so the string values make more sense. Given that a switch will always have 2 values and here making them booleans makes the component less tied to wherever it's getting used, it makes sense. At this point, i think the underlying HTML for this should just be what make the most sense semantically and for keyboard interaction. Which I think is what you have.  That's my 2 cents!

Reply:
I think we're better off keeping it as a radio so that we can always only have one value selected. I updated the values to only call `onChange` with booleans though. Internally their values will be `"ON"` and `"OFF"`

Reply:
I respect your decision to not make it `"TACOS"` & `"ENCHILADAS"` internally